cluster/addons/addon-manager/kube-addons-test.sh

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# These tests enforce behavior of kube-addon-manager functions against a real
+# cluster. A working Kubernetes cluster must be set up with kubectl configured.
+# To run with the released version of kubectl, use `make test`.
+
+set -o errexit
+set -o pipefail
+set -o nounset
+#set -o xtrace
+
+# Default kubectl to the test users installation if needed.
+KUBECTL_BIN="${KUBECTL_BIN:-kubectl}"
+
+# Disabling shellcheck following files as the full path would be required.
+# shellcheck disable=SC1091
+source "kube-addons.sh"
+
+TEST_NS="kube-addon-manager-test"
+
+function setup(){
+  local tries=10
+  while [ "${tries}" -gt 0 ]; do
+    kubectl create namespace "${TEST_NS}" && \
+      return 0;
+    (( tries-- ))
+    sleep 1
+  done
+}
+
+function teardown() {
+  local tries=10
+  while [ "${tries}" -gt 0 ]; do
+    kubectl delete namespace "${TEST_NS}" && \
+      return 0;
+    (( tries-- ))
+    sleep 1
+  done
+}
+
+function error() {
+  echo -e "\e[31m$*\e[0m"
+}
+
+function echo_green() {
+  echo -e "\e[32m$*\e[0m"
+}
+
+function test_create_resource_reconcile() {
```

Comment:
I find it easy to follow the exact test steps when reading the codes, but not as easy when executing it. Could we log necessary info about what the test is doing during the process?